pages/class/[classid].tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useContext } from "react";
+import type { NextApplicationPage } from "../_app";
+import { AuthContext } from "../../context/AuthContext";
+import styles from "../../styles/class.module.css";
+import { Error } from "../../components/util/Error";
+import { Roster } from "../../components/Roster/Roster";
+import { useRouter } from "next/router";
+import { Attendance } from "../../components/Attendance/Attendance";
+import { Module } from "../../components/ModuleView/Module";
+import useSWR from "swr";
+import { APIContext } from "../../context/APIContext";
+
+const BackButton: React.FC = () => {
+  const router = useRouter();
+  return (
+    <>
+      <img src="back_button.svg" onClick={() => router.push("/home")} alt="back" className={styles.backbutton} />
+    </>
+  );
+};
+
+const Class: NextApplicationPage = () => {
+  const router = useRouter();
+  const client = useContext(APIContext);
+
+  const { user } = useContext(AuthContext);
+  const { classid } = router.query;
+  const { data: currClass } = useSWR("/api/class" + classid, () => client.getClass(classid));
+
+  const [attendance, setAttendance] = React.useState(false);
+  const [roster, setRoster] = React.useState(false);
+  const [modules, setModules] = React.useState(false);
+
+  const handleUpdate = (name: string): void => {
+    if (name == "attendance") {
+      setAttendance(true);
+      setRoster(false);
+      setModules(false);
+    } else if (name == "roster") {
+      setAttendance(false);
+      setRoster(true);
+      setModules(false);
+    } else if (name == "modules") {
+      setAttendance(false);
+      setRoster(false);
+      setModules(true);
+    }
+  };
```

Comment:
while this works, it is quite inefficient. We can just do something like `const [currentModule, setCurrentModle = useState<string>("attendance")`. To render, we can use a conditional operator that is basically 
``` javascript 
{ currentModule == "attendance" ? <Attendance/> : currentModule == "roster" ? <Roster/> : <Modules/>
```
I forget what the default should be lol but yeah its better to do it the way above